### PR TITLE
FEAT: Implement MSD Radix Sort

### PR DESCRIPTION
**Pull request by kam-zhan-yue**
To be merged after #82.

## Most Significant Digit (MSD) Radix Sort is here!
Code is clearly outlined in commit messages.
* Added parameters
* Added extra info
* Added instruction
* Added explanation
* Modified pseudocode
* Implemented controller
* Implemented acceptance testing

https://github.com/user-attachments/assets/c3c5aef1-5597-4012-ab50-dfe1a25e7470

### CODE REVIEW

**Comment by Sraccity** [comment]
Kudos: On getting it up

**Nitpick**: Interesting PRs { #82  and #83 }. Can't I just accept this one cause you have the same code in it? What's your reasoning behind two PRs? @kam-zhan-yue

**Comment by kam-zhan-yue** [comment]
@Sraccity Heylo, PRs should aim to introduce or change one thing and should be as small / self-contained as possible. In my opinion, adding a renderer is a new feature and should be treated as its own PR. 

By having multiple, smaller PRs, code becomes easier to read and review (it's even easier when you can read them by commits! which I have done here)

Each commit is its own encapsulated story and is designed for you to review one by one. 

As for why the PRs have the same code, it's due to something called [stacked diffs](https://newsletter.pragmaticengineer.com/p/stacked-diffs). MSD and SRS are both reliant on the Mask Renderer, but they should be in separate PRs. To keep them separate, we rebase MSD and SRS on top of Mask Renderer. Technically, the MSD and SRS PRs should merge into the Mask Renderer branch, but we are dealing with a forked repo, so the branch doesn't exist in the main repo yet.

If you merge Mask Renderer, you will see that MSD and SRS will no longer have that commit (because the relevant code has been merged), and you'll be able to review them in isolation.

TLDR: I interactively rebased and squashed commits into different branches to make things more self-contained.

**Comment by Sraccity** [comment]
@kam-zhan-yue Great reasoning/write-up and yeah I do agree with your reasoning. Having smaller PRs helps with other maintainers to review the improvements done quicker. 

Imma check it once again before merge into the branch

**Review by Sraccity — APPROVED** [review]
**Approved**

Kudos on your PR.

Note: I will delay this merge just to check on the logic and animation part of it. Nothing alarming on the code review part.

Cheers
